include/generator

```diff
@@ -451,18 +461,18 @@ struct __generator_promise_base
         return std::move(__g).get();
     }
 
-    template <std::ranges::range _Rng>
-    __yield_sequence_awaiter<generator<_Ref>> yield_value(std::ranges::elements_of<_Rng> __x) {
+    template <std::ranges::range _Rng, typename _Allocator = use_allocator_arg>
+    __yield_sequence_awaiter<generator<_Ref>> yield_value(std::ranges::elements_of<_Rng, _Allocator> __x) {
         // TODO: Consider propagating parent coroutine's allocator to child generator
         // coroutine here.
-        return yield_value(std::ranges::elements_of([](_Rng&& __rng) -> generator<_Ref> {
+        return yield_value(std::ranges::elements_of([](_Rng&& __rng, allocator_arg_t, _Allocator alloc) -> generator<_Ref, std::remove_reference_t<_Ref>, _Allocator> {
```

Comment:
This could also just be `generator<_Ref, _Value, _Allocator>` to match the type of the parent coroutine - might help reduce the number of template instantiations

Reply:
I added a todo there.
If we want to use the same value type, we'd need the value type in the promise type